### PR TITLE
fix: Prevent slash command popup from activating on invalid inputs

### PR DESCRIPTION
**Pull request by 448523760**
## Slash Command popup issue

#7659

When recalling history, the composer(`codex_tui::bottom_pane::chat_composer`) restores the previous prompt text (which may start with `/`) and then calls `sync_command_popup`. The logic in `sync_command_popup` treats any first line that starts with `/` and has the caret inside the initial `/name` token as an active slash command name:

```rust
let is_editing_slash_command_name = if first_line.starts_with('/') && caret_on_first_line {
    let token_end = first_line
        .char_indices()
        .find(|(_, c)| c.is_whitespace())
        .map(|(i, _)| i)
        .unwrap_or(first_line.len());
    cursor <= token_end
} else {
    false
};
```

This detection does not distinguish between an actual interactive slash command being typed and a normal historical prompt that happens to begin with `/`. As a result, after history recall, the restored prompt like `/ test` is interpreted as an "editing command name" context and the slash-command popup is (re)activated. Once `active_popup` is `ActivePopup::Command`, subsequent `Up` key presses are handled by `handle_key_event_with_slash_popup` instead of `handle_key_event_without_popup`, so they no longer trigger `history.navigate_up(...)` and the session prompt history cannot be scrolled.

### CODE REVIEW

File: codex-rs/tui/src/bottom_pane/chat_composer.rs [hunk 1]
```diff
@@ -1603,7 +1603,24 @@ impl ChatComposer {
...
+            //   when there is no non-whitespace content after the '/' token.
+            //   This prevents recalled texts like "/ test" from activating command mode.
+            // - Otherwise, require that the typed name is a prefix of any built-in command.
+            let looks_like_prefix = if maybe_command_name.is_empty() {
+                rest_after_token.is_empty()
+            } else {
+                built_in_slash_commands()
+                    .into_iter()
+                    .any(|(cmd_name, _)| cmd_name.starts_with(maybe_command_name))
+            };
```
[the remark this answers is not in the document]
Comment (448523760) [reply codex-rs/tui/src/bottom_pane/chat_composer.rs]: good point. will update for custom commands.

File: codex-rs/docs/issue.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Recall Prompt History Issue
```
Comment (etraut-openai) [inline codex-rs/docs/issue.md]: Please remove this file from the PR.

File: codex-rs/tui/src/bottom_pane/chat_composer.rs [hunk 3]
```diff
@@ -3873,4 +3916,75 @@ mod tests {
         assert_eq!(composer.textarea.text(), "z".repeat(count));
         assert!(composer.pending_pastes.is_empty());
     }
+
```
Comment (etraut-openai) [inline codex-rs/tui/src/bottom_pane/chat_composer.rs:3919]: I don't think a test is needed in this case because there's little chance of a regression. We're trying to keep our CI test times from exploding, so we want to add tests only when they matter. Please remove.

**Review by etraut-openai — CHANGES_REQUESTED** [review]

**Comment by etraut-openai** [comment]
Thanks for the contribution. I'm not that familiar with the code you're changing here, so I will ask a colleague to review the change. Before I pass it along for review, please address the comments I added.

File: codex-rs/docs/issue.md (hunk 2 above)
Comment (448523760) [reply codex-rs/docs/issue.md]: accidentally added this file 🤣  removed now~

File: codex-rs/tui/src/bottom_pane/chat_composer.rs (hunk 3 above)
Comment (448523760) [reply codex-rs/tui/src/bottom_pane/chat_composer.rs:3919]: I'm not entirely sure whether these tests are strictly necessary, but the current changes were specifically intended to fix those edge cases. In particular, when the GUI uses the user's history, if a message in the history starts with "/", it can break the GUI's logic. For this kind of fix, I believe adding dedicated unit tests makes sense. I added several messages that start with "/" and a message containing only a single "/" (specially constructed boundary cases) to ensure the GUI handles them correctly.

Of course, if you still think they're unnecessary, I'll remove them. But before doing that, I'd like to get a double confirmation from your side.

**Review by dedrisian-oai — APPROVED** [review]
Thank you for the PR!

**Review by etraut-openai — APPROVED** [review]
